src/main/java/de/eddyson/tapestry/react/ReactModule.java

```diff
@@ -44,11 +47,17 @@
 import de.eddyson.tapestry.react.services.BabelCompiler;
 import de.eddyson.tapestry.react.services.CJSXCompiler;
 import de.eddyson.tapestry.react.services.NodeBabelCompiler;
+import de.eddyson.tapestry.react.services.nashorn.NashornReactRenderEngine;
+import de.eddyson.tapestry.react.services.nashorn.NashornReactRenderEngineImplementation;
 
 public final class ReactModule {
 
   private final static Logger logger = LoggerFactory.getLogger(ReactModule.class);
 
+  public static void bind(ServiceBinder binder) {
+    binder.bind(NashornReactRenderEngine.class, NashornReactRenderEngineImplementation.class);
```

Comment:
Yes, Nashorn should be thread safe. I am not sure if the engine is cumulating heap over time when eval'ing the render commands, going to provide a test for that but still have to figure out how to test. Maybe i am going to render a component 1kk times and compare memory consumption before and after.